### PR TITLE
fix(Date Value Validators): Propagate valueRequiredValidator to child component (DSP-1188)

### PR DESCRIPTION
**Pull request by mdelez**
resolves DSP-1188

### CODE REVIEW

**Comment by mdelez** [comment]
@tobiasschweizer efdd557 Fixes the issue for the date value component. We know we will have this issue for other complex value components as well. Should I fix them in this PR or open a new PR for each component?

P.S. I made a small fix to `int-value-component.spec` in this commit as well 🙊

**Comment by flavens** [comment]
> @tobiasschweizer @flavens [efdd557](https://github.com/dasch-swiss/dsp-ui-lib/commit/efdd55731160c057a7adee0ff5fb3435552f578f) Fixes the issue for the date value component. We know we will have this issue for other complex value components as well. Should I fix them in this PR or open a new PR for each component?

It would be easier to do it in one PR. @tobiasschweizer ?

**Comment by tobiasschweizer** [comment]
One PR seems less complicated.

**Comment by mdelez** [comment]
After talking with @tobiasschweizer about the complexity of all of this, we have decided to split it into one PR per value component to make the reviewing process easier.

**Comment by tobiasschweizer** [comment]
@mdelez I'll do the review one Monday

**Comment by mdelez** [comment]
For the reviewers, to test: 
 - add `[valueRequiredValidator]="false"` to `dsp-date-value` in `add-value.component.html`
 - check that when you click on the add button in the playground for a new date value, you shouldn't see any error messages initially and the save icon should be enabled (although it will fail on knora's end if you click it)
 - change `[valueRequiredValidator]="false"` to true
 - check that the error messages behave correctly and that the save icon is disabled

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts [hunk 1]
```diff
@@ -116,6 +117,26 @@ class TestHostCreateValueComponent implements OnInit {
   }
 }
 
+/**
```
Comment (tobiasschweizer) [inline projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts:120]: Could you also add a test to `projects/dsp-ui/src/lib/viewer/values/date-value/date-input/date-input.component.spec.ts`? Maybe just check for the form to be valid.

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts [hunk 2]
```diff
@@ -611,5 +633,24 @@ describe('DateValueComponent', () => {
...
+      testHostFixture = TestBed.createComponent(TestHostCreateValueNoValueRequiredComponent);
+      testHostComponent = testHostFixture.componentInstance;
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent).toBeTruthy();
+      expect(testHostComponent.inputValueComponent).toBeTruthy();
+    });
+
+    it('should not create an empty value', () => {
+        expect(testHostComponent.inputValueComponent.getNewValue()).toEqual(false);
```
Comment (tobiasschweizer) [inline projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts:651]: You could also a test to check that `valueRequiredValidator` is correctly propagated to `TestDateInputComponent` (binding in `projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.html`).

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts [hunk 3]
```diff
@@ -611,5 +633,24 @@ describe('DateValueComponent', () => {
...
+    beforeEach(() => {
+      testHostFixture = TestBed.createComponent(TestHostCreateValueNoValueRequiredComponent);
+      testHostComponent = testHostFixture.componentInstance;
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent).toBeTruthy();
+      expect(testHostComponent.inputValueComponent).toBeTruthy();
+    });
+
+    it('should not create an empty value', () => {
```
Comment (tobiasschweizer) [inline projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts:650]: Maybe also check that the form is valid.

**Review by tobiasschweizer — COMMENTED** [review]
See my comments on testing.

**Review by flavens — COMMENTED** [review]
I have a question: when you check the box "is a time period", we get an error message to indicate how to fill in the fields. I guess I will get the same problem in my PR if a user checks this box... Is there a possibility to turn it into an hint instead of an error??

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts (hunk 1 above)
Comment (mdelez) [reply projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts:120]: done in c05e0fd

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts (hunk 2 above)
Comment (mdelez) [reply projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts]: doen in c05e0fd

File: projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts (hunk 3 above)
Comment (mdelez) [reply projects/dsp-ui/src/lib/viewer/values/date-value/date-value.component.spec.ts:650]: done in c05e0fd

**Comment by mdelez** [comment]
> I have a question: when you check the box "is a time period", we get an error message to indicate how to fill in the fields. I guess I will get the same problem in my PR if a user checks this box... Is there a possibility to turn it into an hint instead of an error??

I fixed the logic in d113e0b. Thanks for pointing it out :) I kept them as error messages but now they only appear when they should.

**Comment by mdelez** [comment]
> maybe you could add the same test case as above in "'should mark the form's validity correctly'", however this time it is valid.

done in 2f243dd

**Review by tobiasschweizer — APPROVED** [review]
Looks good!

**Pull request merged**